src/dos/program_keyb.cpp

```diff
@@ -20,112 +20,385 @@
-	        "  and code page. It will change to [color=light-cyan]LANGCODE[reset] if provided, optionally with a\n"
-	        "  [color=white]CODEPAGE[reset] and an additional [color=white]CODEPAGEFILE[reset] to load the specified code page\n"
-	        "  number and code page file if provided. This command is especially useful if\n"
-	        "  you use a non-US keyboard, and [color=light-cyan]LANGCODE[reset] can also be set in the configuration\n"
-	        "  file under the [dos] section using the \"keyboardlayout = [color=light-cyan]LANGCODE[reset]\" setting.\n"
+	        "  - Running [color=light-green]keyb[reset] without an argument shows the currently loaded keyboard layout\n"
+	        "    and code page.\n"
+	        "  - The [color=white]CPIFILE[reset], if specified, must contain screen font for the given [color=white]CODEPAGE[reset].\n"
+	        "  - If no custom [color=white]CPIFILE[reset] is specified, it tries to find a suitable screen font\n"
+	        "    in one of the bundled CPI files.\n"
+	        "  - If [color=white]CODEPAGE[reset] is not specified, it tries use the screen font from the display\n"
+	        "    adapter ROM first, just like when using /rom switch.\n"
+	        "  - Only EGA or better display adapters allow to change the screen font; MDA,\n"
+	        "    CGA, or Hercules always use the ROM screen font.\n"
+	        "  - The 'us' keyboard layout can be used with any code page; all the other\n"
+	        "    layouts work with selected code pages only.\n"
 	        "\n"
 	        "Examples:\n"
 	        "  [color=light-green]KEYB[reset]\n"
 	        "  [color=light-green]KEYB[reset] [color=light-cyan]uk[reset]\n"
 	        "  [color=light-green]KEYB[reset] [color=light-cyan]sp[reset] [color=white]850[reset]\n"
 	        "  [color=light-green]KEYB[reset] [color=light-cyan]de[reset] [color=white]858[reset] mycp.cpi\n");
-	MSG_Add("PROGRAM_KEYB_NOERROR","Keyboard layout %s loaded for codepage %i.\n");
-	MSG_Add("PROGRAM_KEYB_FILENOTFOUND","Keyboard file %s not found.\n");
-	MSG_Add("PROGRAM_KEYB_INVALIDFILE","Keyboard file %s invalid.\n");
-	MSG_Add("PROGRAM_KEYB_LAYOUTNOTFOUND","No layout in %s for codepage %i.\n");
-	MSG_Add("PROGRAM_KEYB_INVCPFILE","None or invalid codepage file for layout %s.\n");
+	// Success/status message
+	MSG_Add("PROGRAM_KEYB_CODE_PAGE", "Code page");
+	MSG_Add("PROGRAM_KEYB_ROM_FONT", "ROM font");
+	MSG_Add("PROGRAM_KEYB_CUSTOM_FONT", "custom font");
+	MSG_Add("PROGRAM_KEYB_KEYBOARD_LAYOUT", "Keyboard layout");
+	MSG_Add("PROGRAM_KEYB_KEYBOARD_SCRIPT", "Keyboard script");
+	MSG_Add("PROGRAM_KEYB_NOT_LOADED", "not loaded");
+	// Warnings
+	MSG_Add("PROGRAM_KEYB_WARNING_CODE_PAGE",
+	        "[color=light-red]Warning:[reset] It is recommended to avoid using code page %d!");
+	MSG_Add("PROGRAM_KEYB_WARNING_DOTTED_I",
+	        "It replaces the standard ASCII dotless letter 'I' with a national dotted\n"
```

Comment:
"It replaces" — what is it?

Reply:
The full message looks like on the screenshot. BTW, this code page is insane, and I am seriously considering patching it, so that the ASCII character 0x49 is again the regular uppercase `I`, not a national variant.

![Screenshot](https://github.com/user-attachments/assets/a6925d7c-b35e-4833-8137-af4aabe56d6f)